Vue 3 upgrade

obrienben opened the pull request:
For a high-level review, checking for

- Vue 3 best practices
- Production vs Development practices
- Any obvious flaws, mistakes

Keep in mind that the app is still using the Options API, and not the Vue 3 Composition API. This usage can be refactored later.

obrienben committed:
Start of VUE 3 Migration (July 2025)
==========================
- Removed redundant dependencies from package.json
  - Vue CLI
  - ES Lint (will add back later)
  - Prettier  (will add back later)
  - Babel
  - Node-sass
  - sass-loader
  - vue-template-compiler
- Increase Vue version to 3.5.13
- Replace build tool with Vite
- Update dependency versions to their latest (if compatible)
- Replace vue-sliders with vue-3-sliders
- Rebuilt package-lock.json
- Moved mounting of Vue app to index.html
- Updated scss import statements to '@use'
- Fixed view and component imports. Required .vue extension.
- Updated router configuration
- Fixed default Vue export error with import * as Vue from 'vue'

obrienben committed:
VUE 3 Migration continued (July 2025)
==========================
- Replaced Vuex with Pinia
  - Migrated existing stores to use Pinia, and moved to root of /store directory
  - Components, Views and Mixins updated
    - Import Pinia and relevant stores
    - Replace mapState with mapStores, update mapActions to reference store directly
    - Update use of store state in templates with 'this.<store-name>Store.' prefix
- main.js
  - Configure Vue app to use Pinia and VTooltip
  - Comment out code block that called initializeVue(). Unsure if any of this is still needed.
  - Remove setting of Vue.config.productionTip. Now false by default.
- Disabled LinkGraph.vue and SearchVisualization.vue imports for now due to issues with d3 dependency
- Removed BASE_URL variable use from favicon URL
- Removed redundant solrwayback_index_page.html
- SearchBox component, update use of router to get query string parameters
- Update dependency imports where there is no default export.
  - Replaced 'import <name> from' with 'import * as <name> from'

obrienben committed:
VUE 3 Migration continued (July 2025)
==========================
- Updated NodeJS and NPM versions in POM for build
- Add missing store identifiers to state fields in SearchBox.vue and PrimaryModal.vue
- Add missing 'this' keyword usage in Search Store for state fields
- Set base URL context in Vite config
- Change welcome file config in web.xml to index.html which now has VUE entry point

obrienben committed:
VUE 3 Migration continued (July 2025)
==========================
- Search is now functioning
- Convert PostSearchResults to use dynamic component resolution using defineAsyncComponent
  - Enables search results to render, otherwise component will only render Promise objects
  - SingleEntryComponent switch now uses component objects rather than strings.
- Add more missing store identifiers to state fields in components and mixins
- Replaced call to get query string HistoryRoutingUtils.js with 'this.$route.query'
- Convert RequestService.js to use asynchronous functions
- Convert search.store.js to use asynchronous functions and use the notifier store directly

obrienben committed:
VUE 3 Migration continued (August 2025)
==========================
- Fix dismiss notifications
- Replace remaining uses of this.dispatch
- Add more missing store identifiers to state fields in components
- Replaced more 'this.$router.history.current.query' calls with 'this.$route.query'
- Updated watchers using stores to explicitly reference store properties as required by Pinia
- Router no longer requires 'path:' when binding to 'to:', updated SearchMasonryImage and SearchSingleItemPreviewImage
- Add loading of server config back into main.js
- Re-enabled LinkGraph and SearchVisualization components, and switch Vuex to Pinia
- Switch use of d3 library to a global import, as script tag in main.js. Temporary solution until d3 can be upgraded to a later version.
- Remove some replaced, commented out, code

obrienben committed:
VUE 3 Migration continued (August 2025)
==========================
- Add routes back into router index, /about, /pageharvestdata, /pwid and /linkgraph, and fixed any broken imports in them
- In rewrite.config replace 'solrwayback_index_page.html' with 'index.html'
- Add linting back into project with ESLint and setup basic configuration
- Replace deprecated Vue 2 lifecycle hooks (beforeDestroy)
- Convert Vue 2 filters into methods
- Facade Java class querying incorrect solr field, replaced 'sourceFilePath' with 'source_file_path'
- Remove @vue/compat from package.json

obrienben committed:
VUE 3 Migration continued (August 2025)
==========================
- fireGeoImageSearchRequest incorrectly processing GPS image search result

leefrank9527 commented:
1. The redundant import in notifier.store.js and HarvestCalendar.vue seems can be removed:
` import * as Vue from 'vue': `
Usually, import * as Vue from 'vue' is not considered a best practice in modern Vue 3 projects, except in very specific cases.
2. vuex is still in the package.json. It seems pinia is used in store. Is it possible to remove the dependency from package.json?
3. Some lines end with semi and somes not. Such as:
```
import { defineStore } from 'pinia';
import * as Vue from 'vue';
import { useSearchStore } from '../store/search.store'
```
If we take vs code as the IDE and prettier as code style formatter. We can commit the default settings in the project for convinience. The configuration for WCT UI are:
[](https://github.com/WebCuratorTool/webcurator/blob/features/ui-rebuild/webcurator-ui/.vscode/extensions.json
)
[](https://github.com/WebCuratorTool/webcurator/blob/features/ui-rebuild/webcurator-ui/.prettierrc.json
)
4. If have to refer d3 in the index.html, it would be more compatable by downloading the js file (https://d3js.org/d3.v3.min.js) and put it in the public folder, then import the static js file from index.html.

obrienben committed:
VUE 3 Migration continued (August 2025)
==========================
- Reverted project back to using solrwayback_index_page.html instead of index.html. This is due to the potential collision during web archive playback
  - Configure custom entry point using solrwayback_index_page.html in Vite config
  - Set default Vite Dev and Vite Preview pages to open
  - Add a conditional DEV route to the router, for resolving solrwayback_index_page.html when using Vite Dev and Vite Preview
  - Adjust scripts in package.json to include a build option that uses '--mode preview'
- Removed redundant Vue library import in notifier.store.js and HarvestCalendar.vue
- Remove vuex dependency from package.json
- Include d3.v3.min.js directly in project for now, enables offline use

obrienben committed:
VUE 3 Migration continued (August 2025)
==========================
- main.js updates
  - Make loading of SolrWayback properties more robust by using the application base set in the Vite config
  - Delay mounting of app until after config is loaded

obrienben committed:
VUE 3 Migration continued (August 2025)
==========================
- Previous commit that switched project to use custom entry point was missing solrwayback_index_page.html file

obrienben committed:
VUE 3 Migration continued (August 2025)
==========================
- Remove semi-colon line terminators from Javascript code. To align with convention used in pre Vue 3 codebase

jesperlauridsen committed:
first stab

jesperlauridsen committed:
Should take care of the chart.js and vue-charts.js problems for now.

jesperlauridsen committed:
removed useless debug console.logs

jesperlauridsen committed:
A better way to structure this, I think, without the .env file.

obrienben committed:
Merge branch 'vue-3-vite-setup-for-devel' into vue-3-upgrade
